superset-frontend/src/explore/components/controls/MetricControl/AdhocMetricPopoverTrigger.tsx

```diff
@@ -27,7 +27,10 @@ import { savedMetricType } from './types';
 
 export type AdhocMetricPopoverTriggerProps = {
   adhocMetric: AdhocMetric;
-  onMetricEdit: () => void;
+  onMetricEdit(
+    newMetric: AdhocMetric | Record<string, any>,
+    oldMetric: AdhocMetric | Record<string, any>,
```

Comment:
Is `Record<string, any>` for saved metrics here? Could this be `savedMetricType`? (Unrelated bycatch, this type should probably be called `SavedMetricType`)

Reply:
Maybe `import { Metric } from '@superset-ui/core';` ?

Reply:
<img src="https://user-images.githubusercontent.com/335541/105817925-1ac21b00-5f6b-11eb-821e-e9b25818c22e.png" width="350">

Reply:
`savedMetricType` doesn't include `label` and `hasCustomLabel` fields.